lib/honeybadger/json.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Honeybadger.JSON do
+  def encode(notice) do
+    # try to encode without going the rabbit hole
+    case safe_encode(notice) do
+      {:ok, output} ->
+        {:ok, output}
+
+      {:error, %Protocol.UndefinedError{}} ->
+        notice
+        |> to_encodeable
+        |> Jason.encode()
+    end
+  end
+
+  # struct
+  defp to_encodeable(%_{} = input) do
+    input
+    |> Map.from_struct()
+    |> to_encodeable()
+  end
+
+  # map
+  defp to_encodeable(%{} = input) do
+    for {k, v} <- input, into: %{} do
```

Comment:
We don't use single letter variables in other places, it would be great to avoid them here (and elsewhere in this module) as well.